internal/clients/controller_manager.go

```diff
@@ -17,12 +17,15 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/metrics/server"
 )
 
-func NewControllerManager(config *common.Config, logger logr.Logger) manager.Manager {
+func NewControllerManager(config *common.Config, logger logr.Logger) (manager.Manager, error) {
```

Comment:
A general comment to this function, you can feel to skip it :) 

If you only return error with proper context like

```go
errors.Wrap(cacheOptsErr, "while getting cache option")
```
then you will not need to log them in function body, but you can log them in main function since the error context will finally propagate to main function already. When we add log statements inside each function, sometimes it ends up with duplicated log lines.

Reply:
I changed things a bit, wdyt about the current version?

Reply:
This is also fine to merge, but you could change only the error parts as follows.

```
if cacheOptsErr != nil {
    return nil, errors.Wrap(cacheOptsErr, "failed to get cache options")
}
```

and 

```
if managerErr != nil {
   return nil, errors.Wrap(managerErr, "failed to get manager")
}
```

By doing this, you will be able to understand exact error in main function. This is kind of adding metadata to errors

Reply:
that makes sense, thank you, done!